src/test/java/liuni/LiUniResourceTests.java

```diff
@@ -0,0 +1,163 @@
+        reset(status);
+    }
+
+    /* Test LiUniResource .fetchTimeline() REST method */
+    @Test
+    public void testREST_fetchTimeline_emptyTimeline() {
+        ResponseList<Status> respList = new ResponseListImpl<Status>();
+        try {
+            doReturn(respList).when(twitter).getHomeTimeline();
+
+            Response resp = resource.fetchTimeline();
+
+            assertEquals(resp.getStatus(), Response.Status.OK.getStatusCode());
+            assertEquals(resp.getEntity(), respList);
+        }
+        catch (Exception e) {
+            System.out.println("This exception is not expected.");
+            Assert.fail();
+        }
+    }
+
+    @Test
+    public void testREST_fetchTimeline_updateTimeline() {
+        ResponseList<Status> respList = new ResponseListImpl<Status>();
+        try {
+            doReturn(respList).when(twitter).getHomeTimeline();
+
+            Response resp = resource.fetchTimeline();
+
+            assertEquals(resp.getStatus(), Response.Status.OK.getStatusCode());
+            assertEquals(resp.getEntity(), respList);
+
+            respList.add(status);
+            resp = resource.fetchTimeline();
+
+            assertEquals(resp.getStatus(), Response.Status.OK.getStatusCode());
+            assertEquals(resp.getEntity(), respList);
+        }
+        catch (Exception e) {
+            System.out.println("This exception is not expected.");
+            Assert.fail();
+        }
+    }
+
+    @Test
+    public void testREST_fetchTimeline_fetchTimelineError() {
+        ResponseList<Status> respList = new ResponseListImpl<Status>();
+        respList.add(status);
+        String expectedError = (new ErrorModel()).getGeneralError();
+        try {
+            when(twitter.getHomeTimeline()).thenThrow(new TwitterException(""));
+
+            Response resp = resource.fetchTimeline();
+
+            assertEquals(resp.getStatus(), Response.Status.INTERNAL_SERVER_ERROR.getStatusCode());
+            assertEquals(resp.getEntity().toString(), expectedError);
+        }
+        catch (Exception e) {
+            System.out.println("This exception is not expected.");
+            Assert.fail();
+        }
+    }
+
+    /* Test LiUniResource .postTweet() REST method */
+    @Test
+    public void testREST_postTweet_successfulPost() {
+        String testString = "message";
+        try {
+            when(twitter.updateStatus(testString)).thenReturn(status);
+            when(status.getText()).thenReturn(testString);
+
+            Response resp = resource.postTweet(testString);
+
+            assertEquals(resp.getStatus(), Response.Status.CREATED.getStatusCode());
+            assertEquals(resp.getEntity().toString(), testString);
+        }
+        catch (Exception e) {
+            System.out.println("This exception is not expected.");
+            Assert.fail();
+        }
+    }
+
+    @Test
+    public void testREST_postTweet_badPost_badTweet() {
+        String testString = "";
```

Comment:
use a testString as it's more apparent that it isn't just emptyString that could come back as a default value